files/default/neutron-ha-tool.py

```diff
@@ -559,29 +574,88 @@ def migrate_router(qclient, router_id, agent_id, target_id):
         raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent_id))
+                           (router_id, agent['id']))
 
     # add the router id to a live agent
     router_body = {'router_id': router_id}
-    qclient.add_router_to_l3_agent(target_id, router_body)
+    qclient.add_router_to_l3_agent(target['id'], router_body)
 
     # ensure it is removed or log an error
-    if router_id not in list_routers_on_l3_agent(qclient, target_id):
+    if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
-                           (router_id, agent_id))
+                           (router_id, agent['id']))
+    if wait_for_router:
+        wait_router_migrated(qclient, router_id, target['host'])
+
+
+def wait_router_migrated(qclient, router_id, target_host, maxtries=60):
+    """
+    Returns nothing. Waits for all non-distributed ports and floating IPs
+    of a router for being in the ACTIVE again after a migration.
+
+    :param qclient: A neutron client
+    :param router_id: The id of the router to check
+    :param target_host: The host on which the ports should be ACTIVE
+                        (reflected by the binding:host_id attribute of
+                        a port)
+    :param maxtries: Maximum number of times the ports' status is polled,
+                     an expection is raised there are still ports with a
+                     non ACTIVE status after that.
+    """
+
+    LOG.info("Wait for the ports and floating IPs of router_id=%s "
+             "to be ACTIVE", router_id)
+    remaining_ports = ["dummy"]
+    remaining_fips = ["dummy"]
+    while maxtries:
+        if remaining_ports:
+            router_port_list = qclient.list_ports(device_id=router_id,
+                                                  fields=['id',
+                                                          'status',
+                                                          'binding:host_id',
+                                                          'binding:vif_type'])
+            remaining_ports = [
+                port['id'] for port in router_port_list['ports']
+                if (port['binding:vif_type'] != 'distributed' and
+                    not (port['status'] == 'ACTIVE' and
+                         port['binding:host_id'] == target_host))
+            ]
+            LOG.debug("Ports not ACTIVE on router_id=%s: [%s]",
+                      router_id, ", ".join(remaining_ports))
+        elif remaining_fips:
+            floating_ips = qclient.list_floatingips(router_id=router_id)
+            remaining_fips = [fip['id'] for fip in floating_ips['floatingips']
+                              if fip['status'] != 'ACTIVE']
+            LOG.debug("Floating IPs not active: [%s]",
+                      ", ".join(remaining_fips))
+
+        if remaining_ports or remaining_fips:
```

Comment:
We will no longer need this check I believe, we only need the sleep if the iteration is still non-zero.